metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -381,6 +400,18 @@ func CreateStackdriverSink(uri *url.URL) (core.DataSink, error) {
 		glog.Warning("Cluster location required with new resource model but not provided. Falling back to the zone where Heapster runs.")
 	}
 
+	if clusterLocation == "" {
+		glog.Info("An empty cluster location has been provided, checking the GCE Metadata Server to try to auto-detect.")
+
+		clusterLocation_, err := gce.InstanceAttributeValue("cluster-location")
```

Comment:
The same, shouldn't it be a part of gce.OnGCE() block?

Reply:
Updated